lib/libzfs/libzfs_dataset.c

```diff
@@ -931,8 +931,13 @@ libzfs_mnttab_add(libzfs_handle_t *hdl, const char *special,
 		 * Another thread may have already added this entry
 		 * via libzfs_mnttab_update. If so we should skip it.
 		 */
-		if (avl_find(&hdl->libzfs_mnttab_cache, mtn, NULL) != NULL)
+		if (avl_find(&hdl->libzfs_mnttab_cache, mtn, NULL) != NULL) {
+			free(mtn->mtn_mt.mnt_special);
+			free(mtn->mtn_mt.mnt_mountp);
+			free(mtn->mtn_mt.mnt_fstype);
+			free(mtn->mtn_mt.mnt_mntopts);
 			free(mtn);
+		}
 		else
```

Comment:
nit: can you update the `else` block to use this style:

```c
if () {
       ....
} else {
        avl_add(&hdl->libzfs_mnttab_cache, mtn);
}
```

Reply:
Done.  Sorry for that.